src/main/java/com/radixdlt/crypto/ECKeyPair.java

```diff
@@ -244,4 +244,12 @@ public EncryptedPrivateKey encryptPrivateKeyWithPublicKey(ECPublicKey publicKeyU
 			throw new IllegalStateException("Failed to encrypt `this.privateKey` with provided `ECPublicKey`", e);
 		}
 	}
+
+	public String publicKeyHex() {
+		return Bytes.toHexString(getPublicKey().getBytes());
+	}
+
+	public String privateKeyHex() {
+		return Bytes.toHexString(privateKey);
```

Comment:
I don't think this method should be here, it seems like a security risk.

Reply:
Why is this a security risk? :) It is not :) it's **only** _sugar_ for what is already publically available, this is a public method:

```java
public byte[] getPrivateKey() {
    return this.privateKey;
}
```

So this is just sugar for `Bytes.toHexString(myECKeyPair.getPrivateKey())`, which is verbose but used (and/or will be)  A LOT in tests :).

Since Java doesn't support amazing `extensions` which is possible in Swift, we cannot add these methods to the ECKeyPair class inside our test classes. In Swift I would typically do (inside test target/module):

```swift
extension ECKeyPair {
    var privateKeyHex: String { .. }
}
```

Which would allow for that (computed property (a method like variable, being derived from stored instance variable)) to be visible only inside the test target/module, but the cool thing is that is **_callable on the ECKeyPair instances themselves_**.

In the Java world we would need to retort to ugly global methods like so:

```java
final class KeyFormatter { // inside test package....
    static String privateKeyHexOf(ECKeyPair keyPair) {
        return Bytes.toHexString(keyPair.getPrivateKey())
    }
}
```

But that results in more verbose, less easy to read tests:

```java

assertEquals(expectedPrivateKey, KeyHexFormatter.privateKeyHexOf(ecKeyPair));
```

which is really not more easy to read than just using:
```java
assertEquals(expectedPrivateKey, Bytes.toHexString(keyPair.getPrivateKey());
```

But is still less easy to read what I've made now:

```java
assertEquals(expectedPrivateKey, ecKeyPair.privateKeyHex());
```

Reply:
OK, so if it's only for tests, then make it package-local (remove the public visibility) and annotate with VisibleForTesting.